02_configure_host.sh

```diff
@@ -41,21 +42,61 @@ EOF
     virsh pool-autostart default
 fi
 
-# Allow ipmi to the virtual bmc processes that we just started
-if ! sudo iptables -C INPUT -i baremetal -p udp -m udp --dport 6230:6235 -j ACCEPT 2>/dev/null ; then
-    sudo iptables -I INPUT -i baremetal -p udp -m udp --dport 6230:6235 -j ACCEPT
+# Adding an IP address in the libvirt definition for this network results in
+# dnsmasq being run, we don't want that as we have our own dnsmasq, so set
+# the IP address here
+if [ ! -e /etc/sysconfig/network-scripts/ifcfg-provisioning ] ; then
+    echo -e "DEVICE=provisioning\nTYPE=Bridge\nONBOOT=yes\nNM_CONTROLLED=no\nBOOTPROTO=static\nIPADDR=172.22.0.1\nNETMASK=255.255.255.0" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-provisioning
 fi
+sudo ifdown provisioning || true
+sudo ifup provisioning
 
-#Allow access to dualboot.ipxe
-if ! sudo iptables -C INPUT -p tcp --dport 80 -j ACCEPT 2>/dev/null ; then
-    sudo iptables -I INPUT -p tcp --dport 80 -j ACCEPT
+# Need to pass the provision interface for bare metal
+if [ "$PRO_IF" ]; then
+    echo -e "DEVICE=$PRO_IF\nTYPE=Ethernet\nONBOOT=yes\nNM_CONTROLLED=no\nBRIDGE=provisioning" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-$PRO_IF
+    sudo ifdown $PRO_IF || true
+    sudo ifup $PRO_IF
 fi
 
-#Allow access to tftp server for pxeboot
-if ! sudo iptables -C INPUT -p udp --dport 69 -j ACCEPT 2>/dev/null ; then
-    sudo iptables -I INPUT -p udp --dport 69 -j ACCEPT
+# Create the baremetal bridge
+if [ ! -e /etc/sysconfig/network-scripts/ifcfg-baremetal ] ; then
+    echo -e "DEVICE=baremetal\nTYPE=Bridge\nONBOOT=yes\nNM_CONTROLLED=no" | sudo dd of=/etc/sysconfig/network-scripts/ifcfg-baremetal
+fi
+sudo ifdown baremetal || true
+sudo ifup baremetal
+
+# Add the internal interface to it if requests, this may also be the interface providing
+# external access so we need to make sure we maintain dhcp config if its available
+if [ "$INT_IF" ]; then
```

Comment:
Shouldn't this EXT_IF instead?

Reply:
$INT_IF is the interface that was getting added to the baremetal network (before this patch) which is the behaviour we want to maintain. So $INT_IF would be the correct one to set. 

(we had set $EXT_IF yesterday but this probably only worked because we had manually put the correct interface on the bridge masking our error)